lib/puppet/type/rpmkey.rb

```diff
@@ -23,4 +23,8 @@
     desc "The source of the public key if the key is not already imported."
   end
 
+  autorequire(:file) do
+    self[:source] if self[:source] =~ /^\//
```

Comment:
I am not sure how this behaves when `self[:source]` is `nil` (which makes sense together with `ensure => absent`). On ruby 2.1.5 this does not seem to be an issue, but I am not sure about 1.8.7